R/confusion_matrix.R

```diff
@@ -2,20 +2,66 @@
 #' visual. If predicted_y array is passed in,other evaluation scoring metrics such as
 #' Recall, and precision will also be produced
 #'
-#' @param model trained model
-#' @param X_train a data frame of the training dataset without labels
-#' @param X_test a data frame of the validation dataset without labels
-#' @param y_train a data frame of the training labels
-#' @param y_test a data frame of the validation labels
+#' Implementation based on https://stackoverflow.com/a/59119854
+#'
+#' @param actual_y a data frame of the validation labels
 #' @param predicted_y a data frame of predicted values with model
 #' @param labels a list of string of confusion matrix labels
 #' @param title a string of the confusion matrix label
 #'
-#' @return A plot
+#' @return A plot and a dataframe with scoring metrics
 #' @export
+#' @importFrom magrittr %>%
 #'
 #' @examples
-#' confusion_matrix(svm, X_train, y_train, X_valid, y_valid)
-confusion_matrix <- function(model, X_train, X_test, y_train, y_test, predicted_y = NULL, labels = NULL, title = NULL){
+#' confusion_matrix(as.factor(c("yes", "no", "yes")), as.factor(c("no", "no", "yes")))
+confusion_matrix <- function(actual_y, predicted_y, labels = NULL, title = NULL){
+
+  if (!is.vector(predicted_y) && !is.factor(predicted_y)){
+    stop("Sorry, predicted_y should be a vector or factor.")
+  }
+
+  if (!is.vector(actual_y) && !is.factor(actual_y)){
+    stop("Sorry, actual_y should be a vector or factor.")
+  }
+
+  if (length(actual_y) != length(predicted_y)){
+    stop("Sorry, predicted_y and actual_y should have the same length.")
+  }
+
+  if (is.null(title)) {
+    title = "Confusion Matrix"
+  }
+
+  confusion_matrix <- caret::confusionMatrix(predicted_y, actual_y)
+
+  table <- data.frame(confusion_matrix$table)
+
+
+
 
+  plot_table <- table %>%
+    dplyr::mutate(goodbad = ifelse(table$Prediction == table$Reference, "good", "bad")) %>%
+    dplyr::group_by_(~Reference) %>%
+    dplyr::mutate_(prop = ~Freq/sum(Freq))
+
+
+
+
+
+  confusion_plot <- ggplot2::ggplot(data = plot_table, mapping = ggplot2::aes_string(x = 'Reference', y = 'Prediction', fill = 'goodbad', alpha = 'prop')) +
+    ggplot2::geom_tile() +
+    ggplot2::geom_text(ggplot2::aes_string(label = 'Freq'), vjust = .5, fontface  = "bold", alpha = 1) +
+    ggplot2::scale_fill_manual(values = c(good = "green", bad = "red")) +
+    ggplot2::theme_bw() +
+    ggplot2::xlim(rev(levels(table$Reference))) +
+    ggplot2::theme(legend.position = "none") +
+    ggplot2::labs(title = title)
+
+  metric_score <- broom::tidy(confusion_matrix$byClass)
+
+  print(confusion_plot)
+  return(metric_score)
```

Comment:
I think it maybe worth returning the table as well.
```suggestion
  return(list(cf_table = table, metric = metric_score))
```

Reply:
![image](https://user-images.githubusercontent.com/38637744/76153222-0688dd00-607e-11ea-900b-a0e548261834.png)


that is the table.